### PR TITLE
CB-12730: Compat - INTEGRATE

### PR DESCRIPTION
**Pull request by macdonst**
### Platforms affected

Android

### What does this PR do?

Now that the code included in this plugin has been included in cordova-android master for the 6.3.0 release, the installation of this plugin should be skipped on projects using cordova-android 6.3.0 or greater.

### What testing has been done on this change?

First Project
- Created a version of cordova-plugin-camera that had the dependency tag for cordova-plugin-compat deleted.
- Added the custom cordova-plugin-camera to the project
- Added cordova-android 6.2.3 to the project
- Attempted to build the project, it failed as PermissionHelper was missing
- Added https://github.com/macdonst/cordova-plugin-compat-1 to the project
- Built project, ran it on Android, attempted to take a picture and was asked for permission and the picture was taken successfully

Second Project
- Added the custom cordova-plugin-camera to the project
- Added cordova-android 6.3.0 to the project
- Built project, ran it on Android, attempted to take a picture and was asked for permission and the picture was taken successfully

Third Project
- Added the custom cordova-plugin-camera to the project
- Added cordova-android 6.3.0 to the project
- Attempted to add cordova-plugin-compat 1.2.0 to the project but it was skipped as it won't install on cordova-android 6.3.0 or greater
- Built project, ran it on Android, attempted to take a picture and was asked for permission and the picture was taken successfully

Fourth Project
- Added the custom cordova-plugin-camera to the project
- Added cordova-plugin-compat 1.2.0 to the project
- Added cordova-android 6.3.0 to the project
- Built project, ran it on Android, attempted to take a picture and was asked for permission and the picture was taken successfully

### Checklist
- [x] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [x] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,12 +21,17 @@
...
-    "engines": [
-        {
-            "name": "cordova",
-            "version": ">=5.0.0"
+    "engines": {
+      "cordovaDependencies": {
+        "<1.2.0": {
+          "cordova": ">=5.0.0"
+        },
+        "1.2.0": {
```
Comment (filmaj) [inline package.json]: should this be `>=1.2.0` ?

File: package.json [hunk 2]
```diff
@@ -21,12 +21,17 @@
         "ecosystem:phonegap",
         "cordova-android"
     ],
-    "engines": [
-        {
-            "name": "cordova",
-            "version": ">=5.0.0"
+    "engines": {
+      "cordovaDependencies": {
```
Comment (filmaj) [inline package.json]: can we keep the spacing consistent in the package.json?

**Review by filmaj — APPROVED** [review]
Minor stuff, LGTM otherwise!

Nice testing breakdown.

File: package.json (hunk 1 above)
Comment (macdonst) [reply package.json]: Yes, I suppose it can be. I just assume that 1.2.0 would be the last version of this plugin ever.

File: package.json (hunk 2 above)
Comment (macdonst) [reply package.json]: Sure, I kinda counted on eslint/editorconfig to handle that for me.

**Comment by macdonst** [comment]
@filmaj I made the changes you suggested, tested by creating a new project with cordova-android 6.3.0 and then pushed the fix.

**Comment by stevengill** [comment]
LGTM!

**Pull request closed without merging**